backend/operations_scanner.go

```diff
@@ -736,3 +751,68 @@ func convertNodePoolStatus(nodePoolStatus *arohcpv1alpha1.NodePoolStatus, curren
 
 	return opStatus, opError, err
 }
+
+// convertInflightChecks gets a cluster internal ID, fetches inflight check errors from CS endpoint, and converts them
+// to arm.CloudErrorBody type.
+// The function should be triggered only if inflight errors occurred with provision error code OCM4001.
+func (s *OperationsScanner) convertInflightChecks(ctx context.Context, logger *slog.Logger,
+	internalId ocm.InternalID) (*arm.CloudErrorBody, error) {
+	inflightChecks, err := s.clusterService.GetClusterInflightChecks(ctx, internalId)
+	if err != nil {
+		return &arm.CloudErrorBody{}, err
+	}
+
+	var cloudErrors []arm.CloudErrorBody
+	for _, inflightCheck := range inflightChecks.Items() {
+		if inflightCheck.State() == InflightCheckStateFailed {
+			cloudErrors = append(cloudErrors, convertInflightCheck(inflightCheck, logger))
+		}
+	}
```

Comment:
This _should_ never happen, but maybe we should return an error here if `len(cloudErrors) == 0` after filtering on failed inflight checks... just to ensure we never have a `MultipleErrorsOccurred` with no nested errors.

Reply:
Added.